### PR TITLE
Resolve issues with actor duplication

### LINKED ISSUE #31: Duplicating actors leads to issues with resulting duplicated items
Duplicating actors is a common usage case in order to quickly create similar actors.  Duplicated actors are provided a new unique actorId but all items retain the same itemId.  

If you attempt to delete or modify an item from a duplicated actor it will delete the first item found which matches the itemId instead of the item intended.

This seems to be caused by several functions in [shared.js](https://github.com/free-beer/bh2e/blob/master/module/shared.js) which assume the itemId will be unique when this is not the case.
- _deleteOwnedItem_
- _findItemFromId_
- _findActorFromItemId_

These should target the specific actor instead
for example _deleteOwnedItem_ could be replaced with:
`this.actor.deleteEmbeddedDocuments("Item", [element.dataset.id]);`

Which only deletes the item of the actor which is the current target.

### PR DESCRIPTION
**Pull request by apewall**
This removes relying on the functions mentioned in #31 and uses the current actor object instead.  After some testing it seems to work though I'm not sure it is the best solution.

Other related issues with the character sheet that I'd also like to resolve in this pull but I need some direction of where to look to address them:
- Character sheet tabs not being instantiated which causes tabs on all sheets that are open to change when clicked.
- Clicking buttons on the character sheets (Damage, Repair, Break, Casting) seem to _sometimes_ cause a javascript failure that crashes the browser window.

### CODE REVIEW

**Comment by free-beer** [comment]
I'll take a look through these changes when I get a chance. With regards to the other two issues that you've mentioned, are these things you were seeing before you applied your changes? If so, it's probably better to open issues for them. In relation to the second one specifically can you provide more detail as I personally have never seen that happen?

**Comment by apewall** [comment]
The other problems with the sheets mentioned are not caused by these changes. I'll go ahead and open up new issues with some examples and we can discuss those separately.

**Comment by free-beer** [comment]
So I've thought about this a little more. Given this is such a substantial change I think that to incorporate it it would have to be behind a system setting that is off by default. It complicates things a little but would ensure that people could opt-in to the change.

**Comment by apewall** [comment]
Could you summarize what you are thinking the end-user concern with this change would be?  I assume most are working around the current behavior by avoiding duplicating actors entirely.

**Comment by free-beer** [comment]
Well, the reality is, no-one else has even mentioned this issue. So, if as you say, others are working around this issue, then fundamentally changing this would violate an expectation on their part. By placing this behind an opt-in option then they can knowingly and explicitly change across.

**Comment by apewall** [comment]
I guess I'm failing to understand in what way users would be functionally utilizing the duplication bug.

If they are not already duplicating actors they would not notice this change and remain unaffected in their workflow.  

If they duplicate actors the current behavior is both non-intuitive and destructive since deleting from one actor affects a different actor with a matching itemId instead the one they are actually interacting with.

Ultimately it is up to you, but I don't see a real benefit in making this into an option.

**Comment by yriart** [comment]
Hey, just commenting to say that this bugfix would be helpful for me as well. I can't really imagine how the current state would be preferable since it just makes it so duping sheets doesn't work.
